Add GPU build arg support & Fixes for GPU HW Acceleration

microhobby opened the pull request:
For HW that does not use upstream libs, AKA proprietary GPU Vendor libs, the GPU build arg is usefull to tell the Dockerfile the correct Docker Image to use and which libs install.

Fixes: #51

microhobby committed:
nodeElectron: Bump packages

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
tasks: common: Add GPU build arg for all build container tasks

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
cppQML: Add GPU build arg support

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GPU.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
cppSlint: Add GPU build arg support

This is useful for machines that use proprietary GPU drivers and
libraries.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
python3Pyside2QML: Add GPU build arg support

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GPU.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
python3Pyside2QML: Update to use qtwayland5 and openGL ES

We are installing the openGL ES libs in a multi stage build
and then copying the files to the final image, because pyside2
depends on libqt5gui5 and libqt5quick5, so we are installing
it but overwriting the files with the openGL ES ones.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
python3Pyside2QML: Maintain the consistency of the sample code

We are doing the same of the cppQML sample, so we need to maintain the
consistency of the code.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
nodeElectron: Add the miss wayland as ozone platform for amd64

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
nodeElectron: Add support for GPU build arg

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GP

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

tronical commented on cppSlint/Dockerfile:
```diff
@@ -11,11 +11,17 @@ ARG IMAGE_ARCH=
 ##
 ARG APP_EXECUTABLE=__change__
 
+##
```
Would be awesome if this could also be added to the Slint Rust template. I can also do that after this PR (or make a patch for this branch).

microhobby replied:
Sure, I will add here on my pipeline to include also the `rustSlint` on these changes.

microhobby committed:
nodeElectron: Add suport for EGL Vivante

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
rustSlint: Add GPU build arg support

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GPU.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetAvalonia: Add GPU build arg support

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GPU.

NOTE: The Avalonia GTK uses XWayland, anyway, this should not
work with the Vivante GPU.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetAvaloniaFrameBuffer: Add GPU build arg support

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GPU.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUno: Add GPU build arg support

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GPU.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUno: update to 4.8.33

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUno: Remove openGL flags from launch.json

Uno Platform does not support openGL ES 2.0, so does not make
sense to have this flags in launch.json for now.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

diff --git a/dotnetUno/.vscode/launch.json b/dotnetUno/.vscode/launch.json
index 7617fed..f6c304e 100644
--- a/dotnetUno/.vscode/launch.json
+++ b/dotnetUno/.vscode/launch.json
@@ -8,7 +8,7 @@
       // Use IntelliSense to find out which attributes exist for C# debugging
       // Use hover for the description of the existing attributes
       // For further information visit https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger-launchjson.md
-      "name": "Skia.GTK openGL",
+      "name": "Debug Local",
       "type": "coreclr",
       "request": "launch",
       "preLaunchTask": "build-torizon-local",
@@ -18,28 +18,6 @@
       "env": {
         "DOTNET_MODIFIABLE_ASSEMBLIES": "debug",
         "GDK_BACKEND": "x11",
-        "UNO_DISABLE_OPENGL": "false",
-        "UNO_FULLSCREEN": "false"
-      },
-      "cwd": "${workspaceFolder}/__change__.Skia.Gtk",
-      // For more information about the 'console' field, see https://aka.ms/VSCode-CS-LaunchJson-Console
-      "console": "internalConsole",
-      "stopAtEntry": false
-    },
-    {
-      // Use IntelliSense to find out which attributes exist for C# debugging
-      // Use hover for the description of the existing attributes
-      // For further information visit https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger-launchjson.md
-      "name": "Skia.GTK noGL",
-      "type": "coreclr",
-      "request": "launch",
-      "preLaunchTask": "build-torizon-local",
-      // If you have changed target frameworks, make sure to update the program path.
-      "program": "${workspaceFolder}/__change__.Skia.Gtk/bin/Debug/net6.0/__change__.Skia.Gtk.dll",
-      "args": [],
-      "env": {
-        "DOTNET_MODIFIABLE_ASSEMBLIES": "debug",
-        "UNO_DISABLE_OPENGL": "true",
         "UNO_FULLSCREEN": "false"
       },
       "cwd": "${workspaceFolder}/__change__.Skia.Gtk",
@@ -58,8 +36,6 @@
       "env": {
         "DOTNET_MODIFIABLE_ASSEMBLIES": "debug",
         //"GDK_BACKEND": "x11",
-        //"GDK_GL": "gles",
-        "UNO_DISABLE_OPENGL": "true",
         "UNO_FULLSCREEN": "true"
       },
       "args": [],
@@ -94,7 +70,6 @@
       "env": {
         "DOTNET_MODIFIABLE_ASSEMBLIES": "debug",
         // "GDK_BACKEND": "x11",
-        "UNO_DISABLE_OPENGL": "true",
         "UNO_FULLSCREEN": "true"
       },
       "args": [],
@@ -130,7 +105,6 @@
         "DOTNET_MODIFIABLE_ASSEMBLIES": "debug",
         //"GDK_BACKEND": "x11",
         //"GDK_GL": "gles",
-        "UNO_DISABLE_OPENGL": "true",
         "UNO_FULLSCREEN": "true"
       },
       "args": [],
diff --git a/dotnetUno/__change__.Skia.Gtk/Program.cs b/dotnetUno/__change__.Skia.Gtk/Program.cs
index caf2e7d..cca7b89 100644
--- a/dotnetUno/__change__.Skia.Gtk/Program.cs
+++ b/dotnetUno/__change__.Skia.Gtk/Program.cs
@@ -15,16 +15,8 @@ namespace __change__.Skia.Gtk
 			};

 			var host = new GtkHost(() => new App(), args);
-
-			if (!Environment.GetEnvironmentVariable("UNO_DISABLE_OPENGL")
-				.Equals("true"))
-			{
-				host.RenderSurfaceType = RenderSurfaceType.OpenGL;
-			}
-			else
-			{
-				host.RenderSurfaceType = RenderSurfaceType.Software;
-			}
+			// FIXME: if your machine supports openGL remove this
+			host.RenderSurfaceType = RenderSurfaceType.Software;

 			host.Run();
 		}

microhobby committed:
dotnetUno: Add libgles2 and libgl1-mesa-dev

If we are not using the downstream Vivante GPU drivers we need to
install the mesa drivers to have OpenGL support.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUnoFrameBuffer: Add GPU build arg support

This fixes the build for machines that use a different docker image
with special libs for the specific vendor GPU.

Related-to: #51
Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUnoFrameBuffer: update: Fix .Skia.Linux.FrameBuffer.csproj prefix

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUnoFrameBuffer: update to 4.8.33

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUnoFrameBuffer: Add tty support and attach instead of launch

.NET Uno FrameBuffer template when running from container needs
tty, and also to have it working we need to attach to the process
instead of launch it from ssh, that also redirect the output and we
lost the tty.

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby committed:
dotnetUnoFrameBuffer: update: Add the launch.json

Signed-off-by: Matheus Castello <matheus.castello@toradex.com>

microhobby commented:
Superseded by #67
